compel/src/lib/handle-elf.c

```diff
@@ -16,8 +16,9 @@
 #include "piegen.h"
 #include "log.h"
 
-piegen_opt_t opts = {};
-
+#ifdef CONFIG_MIPS
+#include "arch/mips/src/lib/include/ldsodefs.h"
```

Comment:
This could be simply
`#include "ldsodefs.h" and than you can omit addition `-Icompel` to the `Makefile`:
```
diff --git a/compel/Makefile b/compel/Makefile
index d745d0eb7..de9318c42 100644
--- a/compel/Makefile
+++ b/compel/Makefile
@@ -16,7 +16,6 @@ ldflags-y		+= -r
 #
 # UAPI inclusion, referred as <compel/...>
 ccflags-y		+= -I compel/include/uapi
-ccflags-y		+= -I compel
 
 lib-name		:= $(LIBCOMPEL_A)
 lib-y			+= src/lib/log.o
diff --git a/compel/src/lib/handle-elf.c b/compel/src/lib/handle-elf.c
index ded1908f6..e97d75026 100644
--- a/compel/src/lib/handle-elf.c
+++ b/compel/src/lib/handle-elf.c
@@ -17,7 +17,7 @@
 #include "log.h"
 
 #ifdef CONFIG_MIPS
-#include "arch/mips/src/lib/include/ldsodefs.h"
+#include "ldsodefs.h"
 #endif
 /* Check if pointer is out-of-bound */
 static bool __ptr_oob(const uintptr_t ptr, const uintptr_t start, const size_t size)
```